python/cudf/cudf/core/dataframe.py

```diff
@@ -446,6 +446,14 @@ def __array_function__(self, func, types, args, kwargs):
     def empty(self):
         return not len(self)
 
+    @property
+    def values(self):
+        if not utils._have_cupy:
+            raise ModuleNotFoundError("CuPy was not found.")
+        import cupy
+
+        return cupy.asarray(self.as_gpu_matrix())
```

Comment:
Not needed in this PR, but we should consider just making CuPy a formal dependency instead of optional.

Reply:
It already is formal in cuML, just FYI